File: JAVA/OrdenacaoDePalavrasPorTamanho.java
/**
 * @version JAVA
 * @author MARIO SAM <eu@mariosam.com.br>
 * @see I would love to work with you instead solving web code tests: hire me!
 */
package JAVA;

import java.util.Arrays;
import java.util.Scanner;

public class OrdenacaoDePalavrasPorTamanho {

    public static void main(String[] args) {
		Scanner sc = new Scanner( System.in );
        int n = sc.nextInt();

        for (int i = 0; i < n; i++) {
            String word = sc.next();

            System.out.println( orderWordsBySize(word)) ;
        }
        sc.close();
	}

    /**
     * Return the words ordered by size (desc).
     */
    public static String orderWordsBySize(String words) {
        String[] wordsArray = words.split(" ");
        
        Arrays.sort(wordsArray, (w1, w2) -> w2.length() - w1.length());

        return String.join(" ", wordsArray);
    }
}

File: GO/OrdenacaoDePalavrasPorTamanho.go
/**
 * @version GO
 * @author MARIO SAM <eu@mariosam.com.br>
 * @see I would love to work with you instead solving web code tests: hire me!
 */
package GO

import (
	"sort"
	"strings"
)

/**
 * Return the words ordered by size (desc).
 */
func orderWordsBySize(words string) string {
	wordsArray := strings.Split(words, " ")

	sort.Slice(wordsArray, func(i, j int) bool {
		return len(wordsArray[i]) > len(wordsArray[j])
	})

	return strings.Join(wordsArray, " ")
}
